packages/graphql/src/translate/queryAST/factory/Operations/CreateFactory.ts

```diff
@@ -470,31 +470,31 @@ export class CreateFactory {
         isNested: boolean;
         relationship?: RelationshipAdapter;
     }) {
-        if (!isNested) {
-            entity.getPopulatedByFields("CREATE").forEach((attribute) => {
-                const attachedTo = "node";
-                // the param value it's irrelevant as it will be overwritten by the callback function
-                const callbackParam = new Cypher.Param("1234");
-                const field = new ParamInputField({
-                    attribute,
-                    attachedTo,
-                    inputValue: callbackParam,
-                });
-                create.addField(field);
+        entity.getPopulatedByFields("CREATE").forEach((attribute) => {
+            const attachedTo = "node";
+            // the param value it's irrelevant as it will be overwritten by the callback function
+            const callbackParam = new Cypher.Param("1234");
+            const field = new ParamInputField({
+                attribute,
+                attachedTo,
+                inputValue: callbackParam,
+            });
+            create.addField(field);
 
-                const callbackFunctionName = attribute.annotations.populatedBy?.callback;
-                if (!callbackFunctionName) {
-                    throw new Error(`PopulatedBy callback not found for attribute ${attribute.name}`);
-                }
+            const callbackFunctionName = attribute.annotations.populatedBy?.callback;
+            if (!callbackFunctionName) {
+                throw new Error(`PopulatedBy callback not found for attribute ${attribute.name}`);
+            }
 
-                callbackBucket.addCallback({
-                    functionName: callbackFunctionName,
-                    param: callbackParam,
-                    parent: input,
-                    type: attribute.type,
-                });
+            callbackBucket.addCallback({
+                functionName: callbackFunctionName,
+                param: callbackParam,
+                parent: input,
+                type: attribute.type,
             });
-        } else {
+        });
+
+        if (isNested) {
```

Comment:
I would remove `isNested` from this method and use `relationship` instead.
```suggestion
        if (relationship) {
```